pulpcore/cli/rpm/remote.py

```diff
@@ -43,14 +54,93 @@ def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
+    pulp_option(
+        "--uln-server-base-url",
+        default="https://linux-update.oracle.com/",
+        help=_("ULN Server base URL, default is 'https://linux-update.oracle.com/'"),
+        allowed_with_contexts=(PulpUlnRemoteContext,),
+    ),
+    click.option(
+        "--url",
+        help=_(
+            "For remote_type:uln, Use the ULN channel name starting with uln://. For remote_type:rpm, use url with http/https."
+        ),
+        required=True,
+        callback=_uln_url_callback,
+    ),
+]
+
+remote_create_options = [
+    click.option("--name", required=True),
+    click.option(
+        "--ca-cert",
+        help=_("a PEM encoded CA certificate or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option(
+        "--client-cert",
+        help=_("a PEM encoded client certificate or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option(
+        "--client-key",
+        help=_("a PEM encode private key or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option("--connect-timeout", type=float),
+    click.option(
+        "--download-concurrency", type=int, help=_("total number of simultaneous connections")
+    ),
+    click.option("--password"),
+    click.option("--proxy-url"),
+    click.option("--proxy-username"),
+    click.option("--proxy-password"),
+    click.option("--rate-limit", type=int, help=_("limit download rate in requests per second")),
+    click.option("--sock-connect-timeout", type=float),
+    click.option("--sock-read-timeout", type=float),
+    click.option("--tls-validation", type=bool),
+    click.option("--total-timeout", type=float),
+    click.option("--username"),
+]
+
+remote_update_options = [
+    click.option(
+        "--ca-cert",
+        help=_("a PEM encoded CA certificate or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option(
+        "--client-cert",
+        help=_("a PEM encoded client certificate or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option(
+        "--client-key",
+        help=_("a PEM encode private key or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option("--connect-timeout", type=float),
+    click.option(
+        "--download-concurrency", type=int, help=_("total number of simultaneous connections")
+    ),
+    click.option("--password"),
+    click.option("--proxy-url"),
+    click.option("--proxy-username"),
+    click.option("--proxy-password"),
+    click.option("--rate-limit", type=int, help=_("limit download rate in requests per second")),
+    click.option("--sock-connect-timeout", type=float),
+    click.option("--sock-read-timeout", type=float),
+    click.option("--tls-validation", type=bool),
+    click.option("--total-timeout", type=float),
+    click.option("--username"),
```

Comment:
Now you can merge the `rpm_remote_options` into these two lists with the proper values of `required`.

Reply:
The options list will be repetitive. I can instead create list of  `unique` `remote_create_options`, `remote_update_options`, and another list which have common options for `create/update`. What do you think?

Reply:
Sounds good.

Reply:
Modified the options list.